kedro-datasets/tests/snowflake/test_snowpark_dataset.py

```diff
@@ -1,166 +1,384 @@
 import datetime
-import os
+import sys
+from unittest.mock import MagicMock, patch
 
 import pytest
-from kedro.io.core import DatasetError
-
-try:
-    import snowflake.snowpark as sp
 
-    from kedro_datasets.snowflake import SnowparkTableDataset as spds
-except ImportError:
-    pass  # this is only for test discovery to succeed on Python <> 3.8
+if sys.version_info >= (3, 12):
+    pytest.mark.xfail(
+        "Snowpark is not supported in Python versions higher than 3.11",
+        allow_module_level=True,
+    )
```

Comment:
I did some more testing, and it looks like you can replace this with `snowpark = pytest.importorskip("snowpark")` and it will pass the builds.

Reply:
No way, that's awesome! Let me test :)

Reply:
<img width="678" alt="image" src="https://github.com/user-attachments/assets/acd4e29f-b241-461b-8c9c-495793a4408f">
Doesn't help unfortunately, as it still includes the tests in the coverage..

Reply:
Yeah unfortunately coverage still needs to be skipped, but now at least the tests themselves pass. Before the build were showing:

```
__________ ERROR collecting tests/snowflake/test_snowpark_dataset.py ___________
ImportError while importing test module '/home/runner/work/kedro-plugins/kedro-plugins/kedro-datasets/tests/snowflake/test_snowpark_dataset.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/snowflake/test_snowpark_dataset.py:15: in <module>
    from snowflake.snowpark import DataFrame, Session
E   ModuleNotFoundError: No module named 'snowflake'
=========================== short test summary info ============================
```